envoyauth/response.go

```diff
@@ -422,8 +428,15 @@ func collectEnvoyHeaders(key, value string, result *[]*ext_core_v3.HeaderValueOp
 }
 
 func collectHTTPHeaders(key, value string, result *http.Header) {
+	result.Add(key, value)
+}
+
+func preallocateForEnvoyHeaders(result *[]*ext_core_v3.HeaderValueOption, additional int) {
+	*result = slices.Grow(*result, additional)
+}
+
+func preallocateForHTTPHeaders(result *http.Header, additional int) {
 	if *result == nil {
-		*result = make(http.Header)
+		*result = make(http.Header, additional)
```

Comment:
I think this grows the map itself and not the values for a specific key.
Maybe that is also not easily possiblr

Reply:
Yes, that's true. Will ignore the additional value, but keep for the function signature to match.